2025/week12/tp12_basique.tex

```diff
@@ -280,17 +280,17 @@ \section{Arbre couvrant minimum (ACM) — Minimum Spanning tree (MST)}
         L'algorithme s'arrête car tous les sommets ont été atteints. On voit bien que seules 5 arêtes ont été nécessaires. On remarque que l'ACM n'est pas unique dans ce cas comme on aurait pu choisir l'arête verticale tout à gauche du poids 4 au lieu de l'autre arête du poids 4.
     \end{solution}
 \end{Exercice}
-\begin{Exercice}[15 minutes] \textbf{Algorithme de Kruskal : Python}\\
-    Vous trouverez ci-dessous l'algorithme de Kruskal implémenté en Python (disponible sur Moodle). Parcourez la fonction \textbf{Kruskal\_algo(Graph)} afin de vous assurez que vous ayez bien compris le fonctionnement :\\
+\begin{Exercice}[30 minutes] \textbf{Algorithme de Kruskal\!: Python}\\
+    Vous trouverez ci-dessous l'algorithme de Kruskal implémenté en Python (disponible sur Moodle). Lisez le code afin de vous assurez que vous ayez bien compris le fonctionnement\!:\\
     \lstinputlisting[language=python]{solutions/Question3.py}
     \begin{conseil}
-        L'output de l'algorithme est : \\
-        1 - 2: 2\\
-        2 - 5: 2\\
-        2 - 3: 3\\
-        3 - 4: 3\\
-        0 - 1: 4\\
-        Il se lit comme une liste d'arêtes et de poids à l'arête correspondante.
+        La sortie de l'algorithme est\!: \\
```

Comment:
The images illustrating the graph do not include labels/keys corresponding to the vertices. This may be confusing for students, as the solutions use incremental keys (from 1 to 6) to represent the vertices.